R/metric-selection.R

```diff
@@ -191,6 +192,146 @@ first_eval_time <- function(mtr_set, metric = NULL, eval_time = NULL) {
   summary_res
 }
 
+# ------------------------------------------------------------------------------
+
+#' @rdname choose_metric
+#' @export
+check_metrics_arg <- function(mtr_set, wflow, call = rlang::caller_env()) {
```

Comment:
They are about 3/4 the same but have different use cases. For example, the "arg" function for metrics, sets the defaults if the user offers nothing and so on. 

My initial work about 10 days ago had these two use cases in one function (with subfunctions) and it was too unwieldy. 

Personally, I'd leave them as is but you are absolutely invited to refactor them. Otherwise, I'll take naming candidates :-)